src/modules/AddOnOthers.rb

```diff
@@ -0,0 +1,78 @@
+# encoding: utf-8
+require "yast"
+
+# Yast namespace
+module Yast
+  # This module provides integration of the add-on products
+
+  class AddOnOthersClass < Module
+           
+    include Yast::Logger
+
+    #     add_on_others = [
+    #        {
+    #          "media" => 4, # ID of the source
+    #          "name" : "openSUSE version XX.Y",
+    #          "media_url"=>"dvd:/?devices=/dev/sr1"
+    #          ....
+    #        },
+    #        ...
+    #      ]
+    attr_reader :add_on_others
+
+    def main
+      Yast.import "Pkg"
+      textdomain "packager"      
+
+      @add_on_others = []
+    end
+
+    def Read
+      # Removing all repos which have installed based products
+      # and add-on products.
+      all_products = Pkg.ResolvableProperties("", :product, "")
```

Comment:
`Pkg.ResolvableProperties` should not be used anymore, there is a nice [Resolvable](https://github.com/yast/yast-packager/blob/master/src/lib/y2packager/resolvable.rb#L19-L44) Ruby wrapper.